nicsdru_nidirect_theme.theme

```diff
@@ -50,13 +75,43 @@ function nicsdru_nidirect_theme_preprocess_field(&$variables) {
         break;
     }
   }
-  // Add a class to the anchor tag on application link fields.
+  // Add a class to the anchor tag on application link fields & append the icon
   if (isset($variables['element'])
     && $variables['element']['#field_name'] == 'field_link'
     && $variables['element']['#bundle'] == 'application'
   ) {
-    $classes = ['btn', 'call-to-action'];
-    $variables['items'][0]['content']['#url']->setOption('attributes', ['class' => $classes]);
+    foreach ($variables['items'] as $key => $item) {
+      $icon = '<svg aria-hidden="true" class="ico ico-arrow-right"><use xlink:href="#arrow"></use></svg>';
+      $uri = $item['content']['#title'];
+      $classes = ['btn', 'call-to-action'];
+
+      $formattedIcon = \Drupal\Core\Render\Markup::create($uri . $icon);
+
+      $variables['items'][0]['content']['#title'] = $formattedIcon;
+      $variables['items'][0]['content']['#url']->setOption('attributes', ['class' => $classes]);
+    }
+  }
+  // amend classes on the standard address field output
+  if (isset($variables['element'])
+    && $variables['element']['#field_type'] == 'address'
+  ) {
+    for ($i = 0; $i < count($variables['items']); $i++) {
+      $variables['items'][$i]["content"]["#prefix"] = '<p class="adr" translate="no">';
+      $replacement = array(0 => "street-address");
+      $variables['items'][$i]['content']['address_line1']['#attributes']['class'] = array_replace($variables['items'][$i]['content']['address_line1']['#attributes']['class'], $replacement);
```

Comment:
Not quite sure what the intent is here.

Reply:
removing the default classes applied via the address module and outputting the $replacement as this is what the class should be if adopting microformats architecture

Reply:
That's gonna strip everything out and if that was the intention then you don't need to run array_replace, just assign an array to the 'class' array. If you're looking to replace the core address field classes you'd need to take a different angle.